firmware/src/sensors/lsm6dso32.h

```diff
@@ -0,0 +1,158 @@
+ */
+
+#pragma once
+
+#include <cstdint>
+#include <cstring>
+#include <vector>
+#include "drivers/gpio.h"
+#include "drivers/spi.h"
+#include "util/log.h"
+
+namespace sensors {
+
+class Lsm6dso32 {
+ public:
+  /** Constructor
+   *
+   * @param spi reference the the SPI interface @injected
+   * @param cs reference the the chip select output pin @injected
+   */
+  Lsm6dso32(drivers::Spi& spi, drivers::OutputPin& cs) : m_spi(spi), m_cs(cs) { m_cs.SetHigh(); }
+
+  /** Initialize the sensor
+   *
+   * @return true on success
+   */
+  bool Init() {
+    uint8_t temp;
+    // First check the WHO AM I register to determine if sensor responds
+    ReadRegister(static_cast<uint8_t>(registers::WHO_AM_I), &temp, 1U);
+    if (temp != static_cast<uint8_t>(0x6CU)) {
+      return false;
+    }
+
+    // Configure Accelerometer
+    temp = static_cast<uint8_t>(imu_odr::HZ_104) | static_cast<uint8_t>(accelerometer_fs::G_32);
+    WriteRegister(static_cast<uint8_t>(registers::CTRL1_XL), &temp, 1U);
+
+    // Configure Gyroscope
+    temp = static_cast<uint8_t>(imu_odr::HZ_104) | static_cast<uint8_t>(gyroscope_fs::DPS_2000);
+    WriteRegister(static_cast<uint8_t>(registers::CTRL2_G), &temp, 1U);
+
+    return true;
+  }
+
+  /** Read raw gyroscope data from the sensor
+   *
+   * @param data pointer to write the raw data to, needs to be of size 3!
+   */
+  void ReadGyroRaw(int16_t* data) {
+    ReadRegister(static_cast<uint8_t>(registers::OUTX_L_G), reinterpret_cast<uint8_t*>(data), 6U);
+  }
+
+  /** Read raw accelerometer data from the sensor
+   *
+   * @param data pointer to write the raw data to, needs to be of size 3!
+   */
+  void ReadAccelRaw(int16_t* data) {
+    ReadRegister(static_cast<uint8_t>(registers::OUTX_L_A), reinterpret_cast<uint8_t*>(data), 6U);
+  }
+
+ private:
+  /** Read data from a register of the IMU
+   *
+   * @param reg register to read
+   * @param data pointer to data to store
+   * @param length length of read data
+   */
+  void ReadRegister(const uint8_t reg, uint8_t* const data, const size_t length) {
+    // Set the read flag of the register
+    uint8_t read_reg = reg | static_cast<uint8_t>(0x80U);
+    // Read from the sensor
+    m_cs.SetLow();
+    m_spi.Transfer(&read_reg, 1U);
+    m_spi.Receive(data, length);
+    m_cs.SetHigh();
+  }
+
+  /** Write to a data register of the IMU
+   *
+   * @param reg register to write
+   * @param data data to write
+   * @param length length of write data
+   */
+  void WriteRegister(const uint8_t reg, const uint8_t* data, const size_t length) {
+    // Concatenate the register and the data
+    std::vector<uint8_t> concatenated;
+    concatenated.push_back(reg);
+    concatenated.insert(concatenated.end(), data, data + length);
+    // Transfer the data
+    m_cs.SetLow();
+    m_spi.Transfer(concatenated.data(), length + 1U);
+    m_cs.SetHigh();
+  }
+
+  /// Scoped sensor register enum
+  enum class registers : uint8_t {
```

Comment:
format both the enum name & values like this:

https://google.github.io/styleguide/cppguide.html#Enumerator_Names

Reply:
I feel like this reduces the readability for example:

HZ_12_5 -> kHz125

Which looks like 125kHz instead of 12.5Hz any idea?